cyences_app_for_splunk/default/data/ui/views/cs_malicious_ip_list.xml

```diff
@@ -69,24 +70,42 @@
   <row>
     <panel>
       <title>Globally Detected Malicious IPs</title>
-      <input type="dropdown" token="tkn_ip">
+      <input type="text" token="tkn_ip">
         <label>Search for Bad IP Addresses</label>
+        <default>*</default>
+        <suffix>*</suffix>
+      </input>
+      <input type="dropdown" token="tkn_location">
+        <label>Search By Location</label>
+        <search base="search_list">
+          <query> | search ip=$tkn_ip|s$ description=$tkn_description|s$  |stats count by ip_location</query>
+        </search>
         <choice value="*">All</choice>
```

Comment:
Bro,  in some cases IP location will not be present, so those will not be listed because of condition `iplocation=*` as default. We need to implement something like this.

Ex.
`<input type="text" token="tkn_ip_tmp">
        <label>IP Address</label>
        <default></default>
        <change>
          <condition match="'tkn_ip_tmp'=&quot;&quot;">
            <set token="tkn_ip">()</set>
          </condition>
          <condition>
            <set token="tkn_ip">(ip="*$tkn_ip_tmp$*")</set>
          </condition>
        </change>
      </input>`
      
      See Device Master Table dashboard for reference. See input, init at the top of the dashboard, and tokens in the search queries.